src/test/ui/nll/borrowed-universal-error-2.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(nll)]
+#![allow(warnings)]
+
+fn gimme(x: &(u32,)) -> &u32 { &x.0 }
+
+fn foo<'a>(x: &'a (u32,)) -> &'a u32 {
+    let v = 22;
+    let r = gimme(&(v,));
+    r
+}
```

Comment:
Pushed one more example. Current output:

```bash
> lunch-box. rustc +nightly borrowed-universal-error-2.rs -Znll-dump-cause
error[E0597]: borrowed value does not live long enough
  --> borrowed-universal-error-2.rs:18:20
   |
18 |     let r = gimme(&(v,));
   |                    ^^^^ - temporary value only lives until here
   |                    |
   |                    temporary value does not live long enough
19 |     r
   |     - borrow later used here
   |
   = note: borrowed value must be valid for lifetime '_#3r...

error: aborting due to previous error
```

Reply:
Whoops, this example is not that interesting I guess.  Well, it *is* interesting, but not the reason I initially thought.

Reply:
I thought it would be an example where we are returning something that has to have the lifetime `'a`, like the `borrow-universal-error.rs`, but in fact we get the error earlier on, because of the temporary we've created. Let me revise it a bit.